highlight/vim/syntax/chpl.vim

```diff
@@ -250,26 +258,36 @@ endif
 
 " Chapel extentions
 syn keyword chplStatement	break return continue compilerWarning delete
-syn keyword chplStatement	noinit new delete this these use except only require
+syn keyword chplStatement	new delete this these use except only require
+syn keyword chplStatement	noinit init
 syn keyword chplStatement	as module yield compilerError zip
 syn keyword chplIntent		param type in out inout ref
 syn keyword chplStorageClass    const config export extern var
 syn keyword chplType            domain sparse subdomain range index imag complex int uint real bool
-syn keyword chplType            file string opaque integral numeric enumerated
+syn keyword chplType            file bytes string opaque integral numeric enumerated
 syn keyword chplType            locale sync atomic single dmapped
 syn keyword chplType            owned shared borrowed unmanaged
+syn keyword chplType            nothing void
 syn keyword chplOperator	on reduce scan by align
 syn keyword chplStructure	class record union enum
 syn keyword chplStructure	proc iter cobegin begin local sync let select where
 syn keyword chplStructure	pragma inline with private public forwarding
-syn keyword chplStructure	prototype override
+syn keyword chplStructure	prototype override lifetime
 syn keyword chplBoolean		true false
 syn keyword chplConditional	if then else
-syn keyword chplConstant	nil
+syn keyword chplConstant	nil none
 syn keyword chplRepeat		while for do coforall forall in serial
 syn keyword chplLabel	        when otherwise label
 syn keyword chplErrorHandling   throw throws try catch
 
+" ? is an either ternary condition operator or postfix nilability operator
```

Comment:
`?` isn't a ternary condition operator in Chapel - we spell it out long form:
```
var a = if condition() then thing1() else thing2();
```

`?` can also be used as a query operator:
```
proc f(r: range(?t, ?boundedType, ?stridable)) {}
```